packages/deployment/upgrade-test/Dockerfile

```diff
@@ -1,147 +1,18 @@
-RUN cd upgrade-test-scripts && yarn
-RUN echo '. /usr/src/agoric-sdk/upgrade-test-scripts/env_setup.sh' >> ~/.bashrc
-
-COPY ./${THIS_NAME} ./upgrade-test-scripts/${THIS_NAME}/
-COPY --from=agoric-upgrade-9 /root/.agoric /root/.agoric
-RUN chmod +x ./upgrade-test-scripts/*.sh
-SHELL ["/bin/bash", "-c"]
-RUN . ./upgrade-test-scripts/start_to_to.sh
-
 # UPGRADE
-ARG DEST_IMAGE
-#this is agoric-upgrade-10 upgrading to 11
-#it's a separate target because agoric-upgrade-10 takes so long to test
-FROM ghcr.io/agoric/agoric-sdk:35 as propose-agoric-upgrade-11
-ARG BOOTSTRAP_MODE UPGRADE_INFO_11
-ENV THIS_NAME=propose-agoric-upgrade-11 UPGRADE_TO=agoric-upgrade-11 UPGRADE_INFO=${UPGRADE_INFO_11} BOOTSTRAP_MODE=${BOOTSTRAP_MODE}
-WORKDIR /usr/src/agoric-sdk/
-COPY ./env_setup.sh ./start_to_to.sh ./upgrade-test-scripts/
-RUN cd upgrade-test-scripts && yarn
-RUN echo '. /usr/src/agoric-sdk/upgrade-test-scripts/env_setup.sh' >> ~/.bashrc
-
-COPY ./${THIS_NAME} ./upgrade-test-scripts/${THIS_NAME}/
-COPY --from=agoric-upgrade-10 /root/.agoric /root/.agoric
-RUN chmod +x ./upgrade-test-scripts/*.sh
-SHELL ["/bin/bash", "-c"]
-RUN . ./upgrade-test-scripts/start_to_to.sh
-
-# TEST
-#this is agoric-upgrade-11 / vaults+1
-ARG DEST_IMAGE
-FROM ghcr.io/agoric/agoric-sdk:36 as agoric-upgrade-11
-ARG BOOTSTRAP_MODE
-ENV THIS_NAME=agoric-upgrade-11 BOOTSTRAP_MODE=${BOOTSTRAP_MODE} USE_JS=1
-# start-chain boilerplate
-WORKDIR /usr/src/agoric-sdk/
-COPY ./env_setup.sh ./start_to_to.sh ./package.json ./*.js ./upgrade-test-scripts/
-RUN cd upgrade-test-scripts && yarn
-RUN echo '. /usr/src/agoric-sdk/upgrade-test-scripts/env_setup.sh' >> ~/.bashrc
-
-COPY ./${THIS_NAME} ./upgrade-test-scripts/${THIS_NAME}/
-COPY --from=propose-agoric-upgrade-11 /root/.agoric /root/.agoric
-RUN chmod +x ./upgrade-test-scripts/*.sh
-SHELL ["/bin/bash", "-c"]
-RUN . ./upgrade-test-scripts/start_to_to.sh
-
-# UPGRADE
-ARG DEST_IMAGE
-FROM ghcr.io/agoric/agoric-sdk:36 as propose-agoric-upgrade-12
+FROM ${BASE_IMAGE} as propose-agoric-upgrade-12
 # TODO: Replace with actual Zoe core proposal for upgrade 12 (MCS, Kread, Zoe, restart-contracts, etc)
 ARG BOOTSTRAP_MODE UPGRADE_INFO_12='{"coreProposals":["@agoric/builders/scripts/vats/init-network.js"]}'
-ENV THIS_NAME=propose-agoric-upgrade-12 UPGRADE_TO=agoric-upgrade-12 UPGRADE_INFO=${UPGRADE_INFO_12} BOOTSTRAP_MODE=${BOOTSTRAP_MODE}
-COPY --from=agoric-upgrade-11 /root/.agoric /root/.agoric
-# start-chain boilerplate
-WORKDIR /usr/src/agoric-sdk/
-COPY ./env_setup.sh ./start_to_to.sh ./upgrade-test-scripts/
-RUN cd upgrade-test-scripts && yarn
-RUN echo '. /usr/src/agoric-sdk/upgrade-test-scripts/env_setup.sh' >> ~/.bashrc
+ENV THIS_NAME= UPGRADE_TO=agoric-upgrade-12 UPGRADE_INFO=${UPGRADE_INFO_12} BOOTSTRAP_MODE=${BOOTSTRAP_MODE}
 
-COPY ./${THIS_NAME} ./upgrade-test-scripts/${THIS_NAME}/
-RUN chmod +x ./upgrade-test-scripts/*.sh
+WORKDIR /usr/src/agoric-sdk/
+COPY --chmod=755 ./env_setup.sh ./start_to_to.sh ./upgrade-test-scripts/
```

Comment:
Wondering why the support scripts still need to be in this repo? I though we could get away with only having the test files themselves, and this Dockerfile (and Makefile).

Reply:
The logic to upgrade and then wait for upgrade is `start_to_to.sh`. Test files depends on `env_setup.sh`.

There's more cleanup that can be done but that's outside the scope of this PR.

Reply:
> The logic to upgrade and then wait for upgrade is `start_to_to.sh`. Test files depends on `env_setup.sh`.

But isn't that logic which the base image already needs?

> There's more cleanup that can be done but that's outside the scope of this PR.

Of course, mostly worried about maintenance.